source/ADIOS.cpp

```diff
@@ -25,10 +25,8 @@
 #include "engine/bp/BPFileReader.h"
 #include "engine/bp/BPFileWriter.h"
 
-#ifdef ADIOS_HAVE_DATAMAN // external dependencies
 #include "engine/dataman/DataManReader.h"
```

Comment:
Not really optional, but conditional.  So a user won't be able to turn it on or off, but CMake will determine whether or not to enable it based on whether the platform supports it.  It will work fine from a normal static lib, but when it's built with the -static flag (default for Cray) then you'll get all sorts of warnings regarding how the dlopen functions won't work.